wordpress-mobile/WordPress-Android
# Jetpack Mobile: Scan UiStates part(ii)

Parent https://github.com/wordpress-mobile/WordPress-Android/issues/13326

PS: Size of the PR has increased because of new xml layouts, extracting of styles to a new class. I can split tests (last commit) into a separate PR if it helps in reviewing this PR.

### Description

This PR is a step forward in refining `UiStates` for the `Scan` screen. It mainly includes changes to below classes:

`ScanListItemState` updated to include
- `ScanState` - `ButtonAction` (FixAll, Scan) - (only display logic) do not perform any action yet.
- `ThreatItemState` (corresponding `ThreatViewHolder`, layout etc added)

`ScanStateListItemBuilder`
- Mapping for `ScanState` extracted  in the new `ScanStateListItemBuilder` class, dynamic texts added 

`ScanViewModel`
- `Content` `UiState` updated to include `ScanState` buttons and threats

There will be more states (e.g. after fix action) later, this PR covers only below states, dynamic texts, buttons (displayed when screen is shown) and corresponding tests.
 
 ScanStateModel.State | SCANNING | IDLE | IDLE
-- | -- | -- | --
ScanListItemState  | Scanning | ThreatsNotFound | ThreatsFound (none, some or all fixable)
icon | ic_scan_scanning | ic_scan_idle_threats_not_found | ic_scan_idle_threats_found
title | Preparing to scan | Don\'t worry about a thing | Your site may be at risk
description | Welcome to Jetpack Scan, we are taking a first look at your site now and the result will be with you soon. \n \n We will send you a notification once the scan is complete. In the meantime feel free to continue to use your site as normal. You can check the progress at anytime. | The last Jetpack scan ran _{no. of hours/mins ago or in a few seconds}_ and everything looked great. Run a manual scan now or wait for Jetpack to scan your site later today. | The scan found {no. of threats} potential threats with _{site name or this site}_. Please review them below and take action. We\'re _{here to help}_ if you need us.
fix all button | - | - | Fix all (visibility based on fixable status)
scan button | - | Scan now | Scan again

### To Test

Prerequisite: Make sure both Scan feature flag and MySiteImprovements flag are set to on in the App settings.

- Select a site with scan capability on the My Site tab
- Notice that scan menu is displayed
- Click scan menu item
- Notice that icons, (dynamic) texts,  (dynamic) buttons display as per the table above corresponding to the mentioned states.
- Notice that threats have started displaying (title, description based on threat type yet to be updated)

### Notes

1. Design, icons not final. Ignore all styles.
2. `ScanStatusService` uses `FluxC.ThreatStore` to access threats (only basic details) temporarily just to start displaying the threats, threats count. It needs to be updated later on and so haven't included tests for `ScanStatusService`.
3. "here to help" in the `ThreatsFound` state is not highlighted as a link.

### Merge Instructions

1. Make sure #13578 is merged
2. Ping me for merge. We need to remove fluxc commit hash for getting threats temporarily.

PR submission checklist:
